stubs/aws-xray-sdk/aws_xray_sdk/core/recorder.pyi

```diff
@@ -50,24 +50,24 @@ class AWSXRayRecorder:
         sampler: LocalSampler | DefaultSampler | None = ...,
         stream_sql: bool | None = ...,
     ) -> None: ...
-    def in_segment(self, name: str | None = ..., **segment_kwargs): ...
-    def in_subsegment(self, name: str | None = ..., **subsegment_kwargs): ...
+    def in_segment(self, name: str | None = ..., **segment_kwargs) -> Segment: ...
+    def in_subsegment(self, name: str | None = ..., **subsegment_kwargs) -> Subsegment: ...
     def begin_segment(
         self, name: str | None = ..., traceid: str | None = ..., parent_id: str | None = ..., sampling: bool | None = ...
-    ): ...
+    ) -> Segment | DummySegment: ...
     def end_segment(self, end_time: time.struct_time | None = ...) -> None: ...
-    def current_segment(self): ...
-    def begin_subsegment(self, name: str, namespace: str = ...): ...
-    def current_subsegment(self): ...
+    def current_segment(self) -> Segment: ...
+    def begin_subsegment(self, name: str, namespace: str = ...) -> DummySubsegment | Subsegment | None: ...
+    def current_subsegment(self) -> Subsegment | None: ...
     def end_subsegment(self, end_time: time.struct_time | None = ...) -> None: ...
     def put_annotation(self, key: str, value: Any) -> None: ...
     def put_metadata(self, key: str, value: Any, namespace: str = ...) -> None: ...
-    def is_sampled(self): ...
-    def get_trace_entity(self): ...
+    def is_sampled(self) -> bool: ...
+    def get_trace_entity(self) -> Segment | Subsegment: ...
     def set_trace_entity(self, trace_entity: Context) -> None: ...
```

Comment:
Are these really supposed to have different types?

Reply:
Good spot! Fixed.